### PR TITLE
issue #1844: suggest responses blank if suggester name is not the default

### PR DESCRIPTION
**Pull request by cajahnke**
add optional autocomplete_suggester parameter, in case it has been changed from the default suggested in the Lucene documentation

### CODE REVIEW

File: app/models/concerns/blacklight/suggest/response.rb [hunk 1]
```diff
@@ -2,25 +2,26 @@
...
   module Suggest
     class Response
-      attr_reader :response, :request_params, :suggest_path
+      attr_reader :response, :request_params, :suggest_path, :suggester_name
 
       ##
       # Creates a suggest response
       # @param [RSolr::HashWithResponse] response
       # @param [Hash] request_params
       # @param [String] suggest_path
```
[the remark this answers is not in the document]
Comment (cajahnke) [reply app/models/concerns/blacklight/suggest/response.rb]: oops.  added

**Comment by jcoyne** [comment]
Thanks for your changes.  It appears the `spec/models/blacklight/suggest/response_spec.rb` still needs an update to use the 4 argument method.  Once that is fixed, can you squash your commits?

**Comment by jcoyne** [comment]
You'd be squashing the commits in your own repository, which you already have write access to.   Here's some documentation on how to do this: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by cajahnke** [comment]
Ah, OK. Thanks for the link.

Squashed locally, then pushed the squashed repository back up.

Christopher A Jahnke
Technology Services
Harry Ransom Center
(512) 232-3660

**Comment by cajahnke** [comment]
squashed the commits.  should be ready to go

File: app/models/concerns/blacklight/suggest/response.rb [hunk 2]
```diff
@@ -2,25 +2,27 @@
...
+        @suggester_name = suggester_name
       end
 
       ##
       # Trys the suggestor response to return suggestions if they are
       # present
       # @return [Array]
       def suggestions
-        response.try(:[], suggest_path).try(:[], 'mySuggester').try(:[], request_params[:q]).try(:[], 'suggestions') || []
+        (response.try(:[], suggest_path).try(:[], suggester_name).try(:[], request_params[:q]).try(:[], 'suggestions') || []).uniq
```
Comment (jcoyne) [inline app/models/concerns/blacklight/suggest/response.rb]: Can you tell me why `uniq` is required?

**Comment by cajahnke** [comment]
I mentioned it earlier in case that was a problem – I’m happy to take it out for the purposes of the PR.

It’s for deduplication of suggester response: So, not strictly speaking necessary, but included to address Solr behavior using the default suggester dictionary (see https://stackoverflow.com/questions/30119090/solr-suggester-duplicate-suggestions).  The other dictionary implementations suggest discrete terms instead of the entire field value, which I preferred, but I didn’t like getting one suggestion per document sharing a particular suggested field value, which led to more than half the suggestions being for the same term in some cases.

Since the ‘select’ search is executed without any reference to a prior ‘suggest’ search, there is no difference between picking the first occurrence of a suggested term or the fourth.  I didn’t see any value in seeing multiple instances of a given term in the suggestions and felt the deduplicated results were cleaner.  YMMV

Christopher A Jahnke
Technology Services
Harry Ransom Center
(512) 232-3660

**Comment by jcoyne** [comment]
Thanks for the explanation.

**Pull request merged**